Sources/Bow/Effects/BrightFutures/FutureK.swift

```diff
@@ -0,0 +1,203 @@
+        return value.map(f).k()
+    }
+    
+    public func ap<B>(_ fa : FutureKOf<E, (A) -> B>) -> FutureK<E, B> {
+        return flatMap { a in FutureK<E, (A) -> B>.fix(fa).map { ff in ff(a) } }
+    }
+    
+    public func flatMap<B>(_ f : @escaping (A) -> FutureKOf<E, B>) -> FutureK<E, B> {
+        return value.flatMap { (a : A) -> Future<B, E> in FutureK<E, B>.fix(f(a)).value }.k()
+    }
+    
+    public func handleErrorWith(_ f : @escaping (E) -> FutureKOf<E, A>) -> FutureK<E, A> {
+        return value.recoverWith { e in FutureK<E, A>.fix(f(e)).value }.k()
+    }
+    
+    public func runAsync(_ callback : @escaping (Either<E, A>) -> FutureKOf<E, Unit>) -> FutureK<E, Unit> {
+        return value.flatMap { a in FutureK<E, Unit>.fix(callback(Either.right(a))).value }
+            .recoverWith { e in FutureK<E, Unit>.fix(callback(Either.left(e))).value }.k()
+    }
+}
+
+public extension FutureK {
+    public static func functor() -> FutureKFunctor<E> {
+        return FutureKFunctor<E>()
+    }
+    
+    public static func applicative() -> FutureKApplicative<E> {
+        return FutureKApplicative<E>()
+    }
+    
+    public static func monad() -> FutureKMonad<E> {
+        return FutureKMonad<E>()
+    }
+    
+    public static func applicativeError() -> FutureKApplicativeError<E> {
+        return FutureKApplicativeError<E>()
+    }
+    
+    public static func monadError() -> FutureKMonadError<E> {
+        return FutureKMonadError<E>()
+    }
+    
+    public static func monadDefer() -> FutureKMonadDefer<E> {
+        return FutureKMonadDefer<E>()
+    }
+    
+    public static func async() -> FutureKAsync<E> {
+        return FutureKAsync<E>()
+    }
+    
+    public static func effect() -> FutureKEffect<E> {
+        return FutureKEffect<E>()
+    }
+}
+
+public class FutureKFunctor<E> : Functor where E : Error {
+    public typealias F = FutureKPartial<E>
+    
+    public func map<A, B>(_ fa: FutureKOf<E, A>, _ f: @escaping (A) -> B) -> FutureKOf<E, B> {
+        return FutureK<E, A>.fix(fa).map(f)
+    }
+}
+
+public class FutureKApplicative<E> : FutureKFunctor<E>, Applicative where E : Error {
+    public func pure<A>(_ a: A) -> FutureKOf<E, A> {
+        return FutureK<E, A>.pure(a)
+    }
+    
+    public func ap<A, B>(_ fa: FutureKOf<E, A>, _ ff: FutureKOf<E, (A) -> B>) -> FutureKOf<E, B> {
+        return FutureK<E, A>.fix(fa).ap(ff)
+    }
+}
+
+public class FutureKMonad<E> : FutureKApplicative<E>, Monad where E : Error {
+    public func flatMap<A, B>(_ fa: FutureKOf<E, A>, _ f: @escaping (A) -> FutureKOf<E, B>) -> FutureKOf<E, B> {
+        return FutureK<E, A>.fix(fa).flatMap(f)
+    }
+    
+    public func tailRecM<A, B>(_ a: A, _ f: @escaping (A) -> FutureKOf<E, Either<A, B>>) -> FutureKOf<E, B> {
+        return FutureK<E, A>.tailRecM(a, f)
+    }
+}
+
+public class FutureKApplicativeError<Err> : FutureKApplicative<Err>, ApplicativeError where Err : Error {
```

Comment:
By convention after 0.8.0 Arrow will always have the type class as first type in the extends clause

Reply:
It is not possible in this case as `FutureKApplicative<Err>` is a (super)class, not a protocol. Therefore, in the inheritance clause it must precede any protocol (typeclass in this case).